module/src/components/link/link.component.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+
+import { ClassNames, Globals } from '../..';
+
+interface ILinkPropsCore {
+  to: string;
+  className?: string;
+}
+
+export type ILinkProps<T extends Record<string, any>> = T & ILinkPropsCore;
+
+const DefaultLink: React.FC<ILinkPropsCore> = ({ to, className, children, ...additionalProps }) => (
+  <a {...additionalProps} className={className} href={to}>
+    {children}
+  </a>
+);
+
+interface ILinkContext {
```

Comment:
This essentially introduces the concept of global Armstrong config. Which is a nice thing. But it might be good idea if this piece of context was a global config object that could be extended in the future with other bits of global config?

Something like:
```
interface IArmstrongConfig {
  linkContext: {
    Component: React.FC<ILinkProps<any>>;
    location?: Location | undefined;
    navigate?: (to: string, action: 'push' | 'replace') => void;
  }
}
```
This would prevent us from having to either:
1. Cause a breaking change by renaming this to a config context in the future
2. Keep adding new context and providers for every piece of additional global config to avoid the breaking change, making the provider stack even more unwieldy and nested.

Reply:
Yeah I'm into that

It does reopen the theoretical can of worms of having a single combined context for absolutely everything, but I'm still up for keeping like dialogs and such separate.

We could say for new things that go in context "if it's just some variables that need to be accessible elsewhere, put it in the config context, if its got any logic, give it its own context"?

Reply:
Have moved this to a new ArmstrongConfig context in the latest commits

Reply:
One thought here - the idea is you give `location` a live React state version of location out of, for example, `react-router` which means this'll rerender anything subscribed to ArmstrongConfigContext on any location change which could potentially be undesirable for anything more complex if we're using it for any quite heavy stuff

One to think about down the line, I'm happy with this for now personally